core/templates/pages/exploration-editor-page/feedback-tab/services/thread-data-backend-api.service.spec.ts

```diff
@@ -368,11 +374,13 @@ describe('retrieving threads service', () => {
 
     expect(threadDataBackendApiService.getOpenThreadsCount()).toEqual(0);
     threadDataBackendApiService.createNewThreadAsync(subject, 'Text').then(
-      threadData => {
-        expect(threadData.feedbackThreads.length).toEqual(1);
-        expect(threadData.feedbackThreads[0].threadId)
-          .toEqual('exploration.exp1.jkl1');
-        expect(threadDataBackendApiService.getOpenThreadsCount()).toEqual(1);
+      (value: void | SuggestionAndFeedbackThreads) => {
+        if (value) {
```

Comment:
why the `if` statement? is there any other way to approach this?

Reply:
`value` can be of type void, there might be a case that `feedbackThreads` don't exist on value. Also can't resolve this with `value?.feedbackThreads`.

Reply:
try something like this
```typescript
threadData => {
        const data = <SuggestionAndFeedbackThreads>threadData;
        expect(data.feedbackThreads.length).toEqual(1);
        expect(data.feedbackThreads[0].threadId)
          .toEqual('exploration.exp1.jkl1');
```

Reply:
Done, thanks!